### PR TITLE
Removing --disable-http2 flag and adding --client-protocol flag

### PR DESCRIPTION
**Pull request by sethiay**
This PR includes code changes. I will raise another PR for documentation changes.
Testing details:
1. Manual: 
   - Check client-protocol flag's description and default value using `gcsfuse --help` command.
   - Tried mounting and reading and writing a file with combinations - (http1, jacobsa/gcloud), (http2, jacobsa/gcloud), (http1, Go storage library),  (http2, Go storage library).
   - Build gcsfuse using tools/build_gcsfuse.
   - Mount gcsfuse with http1, http2 protocol using mount command: `mount -t gcsfuse ayushsethi-working-dirs ./gg -o "key_file=key.json,client_protocol=http1,debug_gcs,debug_http,log_file=gg.l"`
   - Tried passing invalid `client-protocol` (http4) and got the expected error.
3. Unit tests:
   - Added unit test case in flags_test.go to test that the input string value for --client-protocol flag is parsed correctly.
   - Added unit test case in flags_test.go to validate the input string values for --client-protocol flag.
   - Made modifications in storage_handle_test.go and main_test.go and ran these tests files manually.
   - CI tests are also passing.
4. Integration tests:
    - Manually ran integration tests in https://github.com/GoogleCloudPlatform/gcsfuse/tree/master/tools/integration_tests/implicitdir. Command used: `GODEBUG=asyncpreemptoff=1 go test -v --testbucket=temp-0011001100-bucket`
5. Perf tests:
    - Read: file size=5mb, client-protocol=http1, go storage client library. 
       - Command used: `gcsfuse --implicit-dirs --max-conns-per-host 100 --client-protocol http1 --only-dir fio_load --key-file ~/key.json ayushsethi-working-dirs gcs`
       - BW: 4143MiB/sec.
     - Read: file size=5mb, go storage client library. (default client protocol)
        - Command used: `gcsfuse --implicit-dirs --max-conns-per-host 100 --only-dir fio_load --key-file ~/key.json ayushsethi-working-dirs gcs`
        - BW: 4033 MiB/sec.
     - Read: file size=5mb, client-protocol=http2, go storage client library
        - Command used: `gcsfuse --implicit-dirs --max-conns-per-host 100 --client-protocol http2 --only-dir fio_load --key-file ~/key.json ayushsethi-working-dirs gcs`
        - BW: 227 MiB/sec.

### CODE REVIEW

File: internal/mount/flag.go [hunk 1]
```diff
@@ -12,28 +12,41 @@
...
 package mount
 
 import "strings"
 
-// Parse an option string in the format accepted by mount(8) and generated for
-// its external mount helpers.
+type ClientProtocol string
+
+const (
+	HTTP1 ClientProtocol = "http1"
```
Comment (vadlakondaswetha) [inline internal/mount/flag.go:22]: I know enums are not directly supported in golang. But can we still use the indirect enum type like this: https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/fs/inode/core.go#L26.
So its more clear what values are expected.

File: perfmetrics/scripts/continuous_test/gcp_ubuntu/build.sh [hunk 2]
```diff
@@ -30,7 +30,7 @@ cd "./perfmetrics/scripts/"
 echo Mounting gcs bucket
 mkdir -p gcs
 LOG_FILE=log-$(date '+%Y-%m-%d').txt
-GCSFUSE_FLAGS="--implicit-dirs --max-conns-per-host 100 --experimental-enable-storage-client-library=true --disable-http2 --debug_fuse --debug_gcs --log-file $LOG_FILE --log-format \"text\" --stackdriver-export-interval=30s"
+GCSFUSE_FLAGS="--implicit-dirs --max-conns-per-host 100 --experimental-enable-storage-client-library=true --debug_fuse --debug_gcs --log-file $LOG_FILE --log-format \"text\" --stackdriver-export-interval=30s"
```
Comment (vadlakondaswetha) [inline perfmetrics/scripts/continuous_test/gcp_ubuntu/build.sh:33]: what is the default value of the flag if nothing is passed?

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

**Review by vadlakondaswetha — COMMENTED** [review]
can you also add testing details in the PR description.

File: internal/mount/flag.go (hunk 1 above)
Comment (sethiay) [reply internal/mount/flag.go:22]: this is similar to the one that you pointed. The only difference is that it is a string enum. 
We can create an int enum and that would require another helper method similar to https://github.com/GoogleCloudPlatform/gcsfuse/blob/421afc5d40dda3976275669f33be2e50f0db550e/internal/fs/inode/core.go#L53 to convert the string value passed in flag into the respective int value of enum element. 
Please let me know if there's any concern around using string enum.

File: perfmetrics/scripts/continuous_test/gcp_ubuntu/build.sh (hunk 2 above)
Comment (sethiay) [reply perfmetrics/scripts/continuous_test/gcp_ubuntu/build.sh:33]: the default value is http1: https://github.com/GoogleCloudPlatform/gcsfuse/blob/1cd60205d80ce4ec286aeda540b7af7c80d302e1/flags.go#L248. 
gcsfuse --help will show that default is http1.

**Review by vadlakondaswetha — APPROVED** [review]

**Review by raj-prince — APPROVED** [review]

**Pull request merged**